pages/sponsor/mint.vue

```diff
@@ -35,6 +35,8 @@ const logger = useLogger();
 const selectedTier = ref<string>('bronze');
 const isApproving = ref<boolean>(false);
 const tokenAllowance = ref<string>('0');
+const approvalType = ref<'unlimited' | 'exact'>('unlimited');
```

Comment:
can you do a 

```typescript
const APPROVAL_TYPE = { UNLIMITED: "unlimited", EXACT: "exact"  } as const
``

 and then get the actual type out of the APPROVAL TYPE and replace the usages of this with the type and the checks with APPROVAL_TYPE.UNLIMITED instead of the string?